tf_agents/agents/dqn/rnddqn_agent.py

```diff
@@ -0,0 +1,456 @@
+      _, _, next_time_steps = self._experience_to_transitions(last_two_steps)
+
+    # Compute RND loss
+    # RND loss should be calculated first, since it is used as intrinsic rewards
+    with tf.name_scope('rnd_loss'):
+      rnd_prediction, _ = self._rnd_network(time_steps.observation,
+                                            time_steps.step_type)
+      rnd_target, _ = self._target_rnd_network(time_steps.observation,
+                                               time_steps.step_type)
+
+      rnd_losses = element_wise_squared_loss(rnd_prediction, rnd_target)
+
+      # if nest_utils.is_batched_nested_tensors(
+      #     time_steps, self.time_step_spec, num_outer_dims=2):
+      #   # Do a sum over the time dimension.
+      #   td_loss = tf.reduce_sum(input_tensor=td_loss, axis=1)
+
+      # Average across the elements of the batch.
+      # Note: We use an element wise loss above to ensure each element is always
+      #   weighted by 1/N where N is the batch size, even when some of the
+      #   weights are zero due to boundary transitions. Weighting by 1/K where K
+      #   is the actual number of non-zero weight would artificially increase
+      #   their contribution in the loss. Think about what would happen as
+      #   the number of boundary samples increases.
+      avg_rnd_loss = tf.reduce_mean(input_tensor=rnd_losses)
+
+      with tf.name_scope('Losses/'):
+        tf.compat.v2.summary.scalar(
+            name='loss', data=avg_rnd_loss, step=self.train_step_counter)
+
+      if self._summarize_grads_and_vars:
+        with tf.name_scope('Variables/'):
+          for var in self._rnd_network.trainable_weights:
+            tf.compat.v2.summary.histogram(
+                name=var.name.replace(':', '_'),
+                data=var,
+                step=self.train_step_counter)
+
+      if self._debug_summaries:
+        common.generate_tensor_summaries('rnd_loss', rnd_losses,
+                                         self.train_step_counter)
+
+    with tf.name_scope('loss'):
+      actions = tf.nest.flatten(actions)[0]
+      q_values, _ = self._q_network(time_steps.observation,
+                                    time_steps.step_type)
+
+      # Handle action_spec.shape=(), and shape=(1,) by using the
+      # multi_dim_actions param.
+      multi_dim_actions = tf.nest.flatten(self._action_spec)[0].shape.ndims > 0
+      q_values = common.index_with_actions(
+          q_values,
+          tf.cast(actions, dtype=tf.int32),
+          multi_dim_actions=multi_dim_actions)
+
+      next_q_values = self._compute_next_q_values(next_time_steps)
+
+      if self._n_step_update == 1:
+        # Special case for n = 1 to avoid a loss of performance.
+        td_targets = compute_td_targets(
+            next_q_values,
+            # TODO Should reward_scale_factor multiply both intrinsic and extrinsic rewards?
+            rewards=reward_scale_factor * (next_time_steps.reward + avg_rnd_loss),
```

Comment:
```suggestion
            rewards=reward_scale_factor * (next_time_steps.reward + rnd_losses),
```

Reply:
Made the change!

Also, I found out that I was using `element_wise_squared_loss()` for RND, which made `rnd_losses.shape == (64, 100)` and `avg_rnd_loss.shape == ()` (64 is the experience batch size, 100 is the final fc layer size). 

I created a new function `mean_squared_loss()`, so now `rnd_losses.shape == (64,)` and `avg_rnd_loss.shape == ()`.